### ISSUE TITLE
Quoting arbitraty string values as Rune source code in a macro

### ISSUE THREAD

**Issue by pkolaczk** [rune-rs/rune#300]
```rust
let value = String::from("A + B + C");
return quote!(#value).into_token_stream();
```

```
error[E0277]: the trait bound `std::string::String: ToTokens` is not satisfied
```

How do I make the rune expression inside the string become a part of code generated by my macro?
The string is received from the user, so I can't know it statically.

**Comment by udoprog** [comment]
You have to pass in a `Lit`, constructed here:
https://docs.rs/rune/0.9.1/rune/ast/enum.Lit.html#method.new

Maybe you wanna [join the Discord?](https://discord.gg/v5AeNkT) There's a channel dedicated to discussing Rune there.

**Comment by pkolaczk** [comment]
Hold on, but wouldn't passing a Lit just emit a single literal?
I mean, I don't want to emit just a String literal "A + B + C".
I want to emit expression A + B + C, that I have in a String.

So finally the code after macro processing would look like this:
```
const FOO = A + B + C;
```

not like this:
```
const FOO = "A + B + C";
```

**Comment by udoprog** [comment]
Ok, I see. Then you do this:
```rust
quote!(A + B + C)
```

And if you want to generate an identifier (e.g. `A` is dynamic) you'd do this:

```rust
let ident = rune::ast::Ident::new("Hello");
quote!(#ident + B + C)
```

**Comment by udoprog** [comment]
After talking on Discord it's apparent that what's needed is the ability to parse a string source to produce a `TokenStream`. This is currently not supported and I've added #302 to track this.

**Comment by udoprog** [comment]
Closed in #302
